### PR TITLE
General XML Dialogue system

### PR DESCRIPTION
**Pull request by VexedProgrammer**
ADD: XML Dialogue system
ADD: Dialogue trigger script
ADD: Personal test scene

### CODE REVIEW

File: Project Rogue/Assets/Scripts/Dialogue/TriggerDialogue.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        if (m_bTriggered) return;
+        if (other.tag == "Player")
+        {
+            m_bTriggered = true;
+            m_Dialogue.Push(m_sSceneName, m_iID, m_fDisplayTime);
+            m_Dialogue.Flush(true);
+        }
+    }
+
+    private void Start()
```
Comment (VexedProgrammer) [inline Project Rogue/Assets/Scripts/Dialogue/TriggerDialogue.cs]: Ah damn I forgot to remove this debug code, I will remove it afterwards haha

**Comment by AlexanderKyron** [comment]
Can you fix up "m_Dialgue" and "m_DiglauePopUp" as well as a few spelling errors in comments?

**Comment by AlexanderKyron** [comment]
See C# naming convention: https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/names-of-type-members

Specifically:

> ❌ DO NOT use a prefix for field names. For example, do not use "g_" or "s_" to indicate static fields.

Could you remove the m_ prefix on member fields? Hungarian notation and other prefixing is generally discouraged in C# (and is only occasionally used for historical reasons as it pops up in very old versions of the .NET API).

**Comment by AlexanderKyron** [comment]
Closed until there is word on using this code as @VexedProgrammer has left the project.

**Pull request closed without merging**